dev/AppLifecycle/AppInstance.cpp

```diff
@@ -566,17 +568,17 @@ namespace winrt::Microsoft::Windows::AppLifecycle::implementation
         auto escapedKey = std::regex_replace(key, std::wregex(L"\\\\"), L"_");
         std::wstring mutexName = wil::str_printf<std::wstring>(L"%s_%s_Mutex", m_moduleName.c_str(), escapedKey.c_str());
 
+        // m_keyCreationMutex is only used to synchronize the owner of the key.  Data access
+        // is still protected by m_dataMutex.
+        // Acquire m_dataMutex before creating m_keyCreationMutex to avoid
+        // processes from seeing the mutex but not seeing the Key on the instance yet.
+        auto releaseOnExit = m_dataMutex.acquire();
+
         // We keep the mutex as a live member to ensure all other instances continue
         // to get an 'open' instead of a 'create' due to it already existing.
-        m_keyCreationMutex.create(mutexName.c_str(), 0, MUTEX_ALL_ACCESS);
-
-        bool currentIsKeyOwner = (GetLastError() != ERROR_ALREADY_EXISTS);
+        bool isCurrentKeyOwner = m_keyCreationMutex.try_create(mutexName.c_str(), 0, MUTEX_ALL_ACCESS);
         if (currentIsKeyOwner)
```

Comment:
Does this compile? The change on 573/579 changes the name of this symbol to `isCurrentKeyOwner` now. Maybe change 579 back to `bool currentIsKeyOwner` ?